site/docs/v1/tech/events-webhooks/signing.adoc

```diff
@@ -0,0 +1,108 @@
+---
+layout: doc
+title: Signing Webhooks
+description: Learn how to sign your Webhook events so you can verify they were sent by FusionAuth
+navcategory: developer
+---
+
+== Signing Webhooks
+
+Signing Webhook events allows you to verify each event was sent by FusionAuth.  Webhook signatures provide an extra layer of security in addition to the methods described in link:/docs/v1/tech/events-webhooks/securing[Securing a Webhook].  Signature verification provides protection against bad actors spoofing webhook event payloads to look like they came from FusionAuth.
+
+This document covers configuring webhook signatures, verifying signatures in your webhook listener, and key rotation.
+
+* <<Configuration>>
+* <<Signature Verification>>
+* <<Example Webhook Listener Code>>
+* <<Testing>>
+* <<Key Rotation>>
+
+=== Configuration
+
+Configuring webhook signatures in FusionAuth consists of generating a key and configuring your webhook to sign using that key.
+
+image::webhook-key-gen.png[Webhook Key Generation,width=1200,role=shadowed]
+
+Keys are generated or imported from [breadcrumb]#Settings -> Key Master#. Webhooks can be signed with three types of keys
+
+* EC key - strongest cryptography, public key can be available
+* RSA key - strong cryptography, public key can be available
+* HMAC key - fast cryptography, requires manual key distribution
+
+EC and RSA keys allow you to make public keys available through the `/.well-known/jwks.json` endpoint, which facilitates key rotation. If your webhook listener cannot make outbound network connections or you prefer to manually configure your key in your webhook listener, HMAC keys are a good option.
+
+For this example, we'll generate an RSA key pair.
+
+Next, you configure your webhook to sign the event with this key. From [breadcrumb]#Settings -> Webhooks#, click on the Edit button for your webhook (or create a new webhook). Select the Signature tab panel. Once you select the check box to enable webhook signatures, a select box allows you to choose the generated key.
```

Comment:
Based on some other examples, we use `breadcrumb` when referring to tabs in the UI. I would recommend changing up how this refers to UI elements because the enable is not a checkbox, and the select is a dropdown ("select box" sounds to me like the redirect URI fields that you can enter and add multiple values).

```suggestion
Next, you configure your webhook to sign the event with this key. From [breadcrumb]#Settings -> Webhooks#, click on the Edit button for your webhook (or create a new webhook). Select the [breadcrumb]#Signature# tab panel. Once you enable webhook signatures, a select dropdown allows you to choose the generated key.
```